sdk/src/client/api/block_builder/transaction_builder/mod.rs

```diff
@@ -40,14 +43,133 @@ use crate::{
     },
 };
 
-/// Working state for the input selection algorithm.
+impl Client {
+    /// Builds a transaction using the given inputs, outputs, addresses, and options.
+    pub async fn build_transaction(
```

Comment:
Wait, why is this on the client?

Reply:
Because it can be :)

Reply:
Tbh not sure if we want this here, it's a bit limited and might be annoying to maintain
Would rather put this in an example

Reply:
how is it limited?

Reply:
Because just the addresses are provided there is `chain: None,`, so after building the user has to know which input needs to be signed with what

Reply:
Yes it's new

Reply:
is there another way we could provide that, you think?

Reply:
Together with the address is probably the best, but even then I'm not sure about this. If users want to do something with foundries , they would also need to get the outputs themselves first to get account outputs, so they can get their addresses and provide it
And there might be more things

Reply:
Depends on how flexible you want it to be :shrug:

Reply:
I mean the wallet only uses the one chain so

Reply:
Kinda agree with Thoralf on this, not sure it's super useful and in general I would add things only when necessary/asked. I don't have a strong opinion though.

Reply:
Really though if we provide the chain similarly to how the wallet has one then the functionality will be identical right?